src/blog/2024/04/building-an-admin-panel-in-node-red-with-dashboard-2.md

```diff
@@ -0,0 +1,111 @@
+
+<!--more-->
+
+If you're new to Dashboard 2.0, refer to our blog post [Getting Started with Dashboard 2.0](https://flowfuse.com/blog/2024/03/dashboard-getting-started/), and Dashboard 2.0 to get things started.
+
+## Enabling FlowFuse User Authentication
+
+Before proceeding further, let’s enable FlowFuse user authentication. This step adds an extra layer of protection to our dashboard by adding a login page that restricts access exclusively to registered FlowFuse users. Additionally, it further simplifies the process for the FlowFuse Multiuser addon to track logged-in users on the dashboard.
+
+1. Navigate to the instance setting.
+2. Select the “Security” tab.
+3. Enable “FlowFuse User Authentication”
+
+!["Screenshot displaying the configuration settings within the FlowFuse instance, enabling user authentication for enhanced security.
+"](./images/building-admin-panel-node-red-dashboard-2-flowfuse-instance-setting.png "Screenshot displaying the configuration settings within the FlowFuse instance, enabling user authentication for enhanced security.
+"){data-zoomable}
+
+## Installing FlowFuse Multiuser Addon 
+
+The FlowFuse Multiuser Addon is a plugin developed for Dashboard 2.0 to access logged-in user data on the dashboard. For more information on FlowFuse Multiuser Addon refer to [Exploring the FlowFuse User Addon ](https://flowfuse.com/blog/2024/04/displaying-logged-in-users-on-dashboard/#Enabling-FlowFuse-User-Authentication)
+
+1. Click the Node-RED Settings (top-right)
+2. Click "Manage Palette"
+3. Switch to the "Install" tab
+4. Search for `@flowfuse/node-red-dashboard-2-user-addon`
+
+##  Building Admin-only page
+
+### Outlining approach 
+
+Before we start building the Admin-only page, let's outline our approach :
+
+- Once the FlowFuse Multiuser addon is configured, each Dashboard 2.0 node emits a `msg` which will have user information attached as `msg._client.user`.
+- We'll store the list of admin user’s usernames in the Node-RED global context.
+- Using an event node, we'll track visitors to the admin-only page.
+- If the visitor's username is contained in the list of admin usernames, they'll see the page; otherwise, the page will be hidden.
+
+### Practical implementation
+
+Now, let's proceed with the practical steps to implement the admin-only page:
+
+
+- Create an admin page in Dashboard 2.0 to hide sensitive data from regular users.
+- Drag an event node on the canvas, then click on it, and choose the UI base of which your admin page and other pages are part.
+- Drag a switch node on the canvas, and add two one to check whether the user’s username is contained in the admin list or a second for otherwise, see the below image.
```

Comment:
As these are steps that need to be processed in order, a numbered list is better than a bullet point list.

```suggestion
2. Drag an event node on the canvas, then click on it, and choose the UI base of which your admin page and other pages are part.
3. Drag a switch node on the canvas, and add two one to check whether the user’s username is contained in the admin list or a second for otherwise, see the below image.
```

Reply:
I apologize for the mistake on my part, but thank you for bringing it to my attention.